apps/api/src/app/events/services/workflow-queue/workflow.queue.service.ts

```diff
@@ -126,21 +124,31 @@ export class WorkflowQueueService {
       removeOnComplete: true,
       removeOnFail: true,
       delay,
-      backoff: {
-        type: 'webhookFilterBackoff',
-      },
     };
 
+    const stepContainsWebhookFilter = this.stepContainsFilter(data, 'webhook');
+
+    if (stepContainsWebhookFilter) {
+      options.backoff = {
+        type: 'webhookFilterBackoff',
+      };
+      options.attempts = this.DEFAULT_ATTEMPTS;
+    }
+
     await this.queue.add(id, data, options);
   }
 
+  private stepContainsFilter(data: JobEntity, onFilter: string) {
+    return data.step.filters.some((filter) => {
```

Comment:
```suggestion
    return data.step.filters?.some((filter) => {
```

I think this causes the tests to fail, when no filters are available on a step

Reply:
Yeah, it is part of the issue I already have it locally but the test still fails, I am looking into it maybe there is another issue.

Reply:
Nope it probably was this one (the confusion on my side was due to the failing localstack locally +  jobs from the previous run that was not handled)